src/main/java/com/cburch/logisim/gui/start/Startup.java

```diff
@@ -145,388 +155,534 @@ static void doPrint(File file) {
+    // FIXME: Shouldn't we exit here; -> RC.QUIT
+    return RC.WARN;
+  }
 
-    if (ret.exitAfterStartup && ret.filesToOpen.isEmpty()) {
-      printUsage();
+  private static RC handleArgSubstitute(Startup startup, Option opt) {
+    final var fileA = new File(opt.getValues()[0]);
+    final var fileB = new File(opt.getValues()[1]);
+    if (!startup.substitutions.containsKey(fileA)) {
+      startup.substitutions.put(fileA, fileB);
+      return RC.OK;
     }
-    if (ret.isTty && ret.filesToOpen.isEmpty()) {
-      logger.error("{}", S.get("ttyNeedsFileError"));
-      return null;
+
+    // FIXME: warning should be sufficient here maybe?
+    logger.error(S.get("argDuplicateSubstitutionError"));
+    return RC.QUIT;
+  }
+
+  private static RC handleArgLoad(Startup startup, Option opt) {
+    if (startup.loadFile != null) {
+      logger.error(S.get("loadMultipleError"));
+      // FIXME: shouldn't we quit here? -> RC.QUIT;
+      return RC.WARN;
     }
-    if (ret.loadFile != null && !ret.isTty) {
-      logger.error("{}", S.get("loadNeedsTtyError"));
-      return null;
+    final var fileName = opt.getValue();
+    startup.loadFile = new File(fileName);
+    return RC.OK;
+  }
+
+  private static RC handleArgEmpty(Startup startup, Option opt) {
+    if (startup.templFile != null || startup.templEmpty || startup.templPlain) {
+      logger.error(S.get("argOneTemplateError"));
+      return RC.QUIT;
     }
+    startup.templEmpty = true;
+    return RC.OK;
+  }
 
-    return ret;
-  }
-
-  private static void printUsage() {
-    System.err.println(S.get("argUsage", Startup.class.getName()));
-    System.err.println();
-    System.err.println(S.get("argOptionHeader"));
-    String[] opts = {
-      "argGeometryOption",
-      "argAccentsOption",
-      "argClearOption",
-      "argEmptyOption",
-      "argTestOption",
-      "argGatesOption",
-      "argHelpOption",
-      "argLoadOption",
-      "argLocaleOption",
-      "argNoSplashOption",
-      "argPlainOption",
-      "argSubOption",
-      "argTemplateOption",
-      "argTtyOption",
-      "argQuestaOption",
-      "argVersionOption",
-      "argTestCircGen",
-      "argTestCircuit",
-      "argTestImplement",
-      "argCircuitOption",
-    };
-    for (final var opt : opts) {
-      System.err.println("   " + S.get(opt));
+  private static RC handleArgPlain(Startup startup, Option opt) {
+    if (startup.templFile != null || startup.templEmpty || startup.templPlain) {
+      logger.error(S.get("argOneTemplateError"));
+      return RC.QUIT;
+    }
+    startup.templPlain = true;
+    return RC.OK;
+  }
+
+  private static RC handleArgGates(Startup startup, Option opt) {
+    final var gateShape = opt.getValue();
+    if (gateShape.equals("shaped")) {
+      AppPreferences.GATE_SHAPE.set(AppPreferences.SHAPE_SHAPED);
+      return RC.OK;
+    } else if (gateShape.equals("rectangular")) {
+      AppPreferences.GATE_SHAPE.set(AppPreferences.SHAPE_RECTANGULAR);
```

Comment:
Maybe change "rectangular" into "IEC"

Reply:
This needs also be separate effort. As we need to rename constants, update help string, localization etc. You can create separate ticket for this change though.